src/IceRpc.Transports.Quic/Internal/QuicMultiplexedConnection.cs

```diff
@@ -89,6 +89,11 @@ public async ValueTask<IMultiplexedStream> CreateStreamAsync(
         {
             throw exception.ToIceRpcException();
         }
+        catch (ObjectDisposedException)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            throw;
```

Comment:
This additional fix is to ensure QUIC and Slic behave the same. CreateStreamAsync throw OperationCanceledException if ODE is received and task is already canceled.